### ISSUE TITLE
ClassCastException in Android Studio preview screen.

### ISSUE THREAD

**Issue by mrz** [journeyapps/zxing-android-embedded#56]
The following exception is thrown every time I display the preview of a layout containing a `com.journeyapps.barcodescanner.CompoundBarcodeView` in Android Studio 1.2.

```
java.lang.ClassCastException: com.android.layoutlib.bridge.android.BridgeContext cannot be cast to android.app.Activity
	at com.journeyapps.barcodescanner.CameraPreview.initialize(CameraPreview.java:173)
	at com.journeyapps.barcodescanner.CameraPreview.<init>(CameraPreview.java:158)
	at com.journeyapps.barcodescanner.BarcodeView.<init>(BarcodeView.java:80)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:408)
	at org.jetbrains.android.uipreview.ViewLoader.createNewInstance(ViewLoader.java:413)
	at org.jetbrains.android.uipreview.ViewLoader.loadView(ViewLoader.java:105)
	at com.android.tools.idea.rendering.LayoutlibCallback.loadView(LayoutlibCallback.java:177)
	at android.view.BridgeInflater.loadCustomView(BridgeInflater.java:214)
	at android.view.BridgeInflater.createViewFromTag(BridgeInflater.java:142)
	at android.view.LayoutInflater.rInflate_Original(LayoutInflater.java:806)
	at android.view.LayoutInflater_Delegate.rInflate(LayoutInflater_Delegate.java:64)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:782)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:479)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:385)
	at android.view.BridgeInflater.inflate(BridgeInflater.java:192)
	at android.view.View.inflate(View.java:18532)
	at com.journeyapps.barcodescanner.CompoundBarcodeView.initialize(CompoundBarcodeView.java:71)
	at com.journeyapps.barcodescanner.CompoundBarcodeView.<init>(CompoundBarcodeView.java:62)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:408)
	at org.jetbrains.android.uipreview.ViewLoader.createNewInstance(ViewLoader.java:413)
	at org.jetbrains.android.uipreview.ViewLoader.loadView(ViewLoader.java:105)
	at com.android.tools.idea.rendering.LayoutlibCallback.loadView(LayoutlibCallback.java:177)
	at android.view.BridgeInflater.loadCustomView(BridgeInflater.java:214)
	at android.view.BridgeInflater.createViewFromTag(BridgeInflater.java:142)
	at android.view.LayoutInflater.rInflate_Original(LayoutInflater.java:806)
	at android.view.LayoutInflater_Delegate.rInflate(LayoutInflater_Delegate.java:64)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:782)
	at android.view.LayoutInflater_Delegate.parseInclude(LayoutInflater_Delegate.java:154)
	at android.view.LayoutInflater.parseInclude(LayoutInflater.java:858)
	at android.view.LayoutInflater.rInflate_Original(LayoutInflater.java:802)
	at android.view.LayoutInflater_Delegate.rInflate(LayoutInflater_Delegate.java:64)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:782)
	at android.view.LayoutInflater.rInflate_Original(LayoutInflater.java:809)
	at android.view.LayoutInflater_Delegate.rInflate(LayoutInflater_Delegate.java:64)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:782)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:504)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:385)
	at com.android.layoutlib.bridge.impl.RenderSessionImpl.inflate(RenderSessionImpl.java:413)
	at com.android.layoutlib.bridge.Bridge.createSession(Bridge.java:321)
	at com.android.ide.common.rendering.LayoutLibrary.createSession(LayoutLibrary.java:350)
	at com.android.tools.idea.rendering.RenderTask$2.compute(RenderTask.java:497)
	at com.android.tools.idea.rendering.RenderTask$2.compute(RenderTask.java:485)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:894)
	at com.android.tools.idea.rendering.RenderTask.createRenderSession(RenderTask.java:485)
	at com.android.tools.idea.rendering.RenderTask.render(RenderTask.java:590)
	at org.jetbrains.android.uipreview.AndroidLayoutPreviewToolWindowManager.doRender(AndroidLayoutPreviewToolWindowManager.java:644)
	at org.jetbrains.android.uipreview.AndroidLayoutPreviewToolWindowManager.access$1700(AndroidLayoutPreviewToolWindowManager.java:79)
	at org.jetbrains.android.uipreview.AndroidLayoutPreviewToolWindowManager$7$1.run(AndroidLayoutPreviewToolWindowManager.java:586)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:152)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:452)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:402)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:137)
	at org.jetbrains.android.uipreview.AndroidLayoutPreviewToolWindowManager$7.run(AndroidLayoutPreviewToolWindowManager.java:581)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:320)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:310)
	at com.intellij.util.ui.update.MergingUpdateQueue$2.run(MergingUpdateQueue.java:254)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:269)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:227)
	at com.intellij.util.ui.update.MergingUpdateQueue.run(MergingUpdateQueue.java:217)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:351)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by rkistner** [comment]
Thanks, will look at this when I have some time. The solution is probably to just disable the camera for the layout preview.

**Comment by rkistner** [comment]
Fix released in v3.0.1.

**Comment by mrz** [comment]
Awesome work, I can confirm it's working now, thanks a lot :).
